source/read/change-streams.txt

```diff
@@ -0,0 +1,261 @@
+   :end-before: end-open-change-stream
+   :language: php
+   :dedent:
+
+To begin watching for changes, run the preceding code. Then, in a separate
+application or shell, modify the ``restaurants`` collection. The following
+example updates a document that has a ``name`` field value of ``'Blarney Castle'``:
+
+.. _php-change-stream-update:
+
+.. literalinclude:: /includes/read/change-streams.php
+   :start-after: start-update-for-change-stream
+   :end-before: end-update-for-change-stream
+   :language: php
+   :dedent:
+
+When you update the collection, the change stream application prints the change
+as it occurs. The printed change event resembles the following output:
+
+.. code-block:: none
+   :copyable: false
+
+   {"_id":{"_data":"..."},"operationType":"update","clusterTime":{"$timestamp":{...},
+   "wallTime":{"$date":...},"ns":{"db":"sample_restaurants","coll":"restaurants"},
+   "documentKey":{"_id":{"$oid":"..."}},"updateDescription":{"updatedFields":
+   {"cuisine":"Irish"},"removedFields":[],"truncatedArrays":[]}}}
+
+Modify the Change Stream Output
+-------------------------------
+
+To modify the change stream output, you can pass pipeline stages in an array as a
+parameter to the ``watch()`` method. You can include the following stages in the
+array:
+
+- ``$addFields`` or ``$set``: Adds new fields to documents
+- ``$match``: Filters the documents
+- ``$project``: Projects a subset of the document fields
+- ``$replaceWith`` or ``$replaceRoot``: Replaces the input document with the
+  specified document
+- ``$redact``: Restricts the contents of the documents
+- ``$unset``: Removes fields from documents
+
+The following passes a pipeline that includes the ``$match`` stage to the ``watch()``
+method. This instructs the ``watch()`` method to output events only when update
+operations occur:
+
+.. literalinclude:: /includes/read/change-streams.php
+   :start-after: start-change-stream-pipeline
+   :end-before: end-change-stream-pipeline
+   :language: php
+   :dedent:
+
+Modify ``watch()`` Behavior
+---------------------------
+
+To modify the behavior of the ``watch()`` method, you can pass an options array
+as a parameter to ``watch()``. The following table describes useful options you
+can set in the array:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Option
+     - Description
+
+   * - ``fullDocument``
+     - | Specifies whether to show the full document after the change, rather
+         than showing only the changes made to the document. To learn more about
+         this option, see the :ref:`php-change-stream-pre-post-image` section of this
+         guide.
+
+   * - ``fullDocumentBeforeChange``
+     - | Specifies whether to show the full document as it was before the change, rather
+         than showing only the changes made to the document. To learn more about
+         this option, see :ref:`php-change-stream-pre-post-image`.
+    
+   * - ``resumeAfter``
```

Comment:
Note: I assume this change stream tutorial is common to other driver docs, so consider this a more general question beyond the scope of PHPLIB.

Is `resumeAfter` actually relevant to the users? I know it's the original option and used internally by drivers when resuming, but my understanding is that `startAfter` is preferred for users.

Looking at [`db.collection.watch()`](https://www.mongodb.com/docs/manual/reference/method/db.collection.watch/), which documents both options, it looks like `resumeAfter` only differs by its limitation to _not_ allow resuming after an "invalidate" event. Given that, it doesn't seem like a "useful option" to include in this table (although it absolutely belongs in the API reference).

Reply:
If startAfter is preferred, I'm happy to remove resumeAfter